index.html

```diff
@@ -5980,94 +5977,84 @@ <h2 id="AudioWorklet-Examples">
   }
 
   draw () {
-    // Draw the meter based on the volume value.
+    /* Draw the meter based on the volume value. */
   }
 
-  // handle updated values from audio side
-  onmessage (event) {
-    if (event.data.hasOwnProperty('volume'))
-      this._volume = event.data.volume;
-  }
 }
 
-var importAudioWorkletNode = window.audioWorklet.import('vumeterprocessor.js');
+// The application can use the node when this promise resolves.
+let importAudioWorkletNode = window.audioWorklet.import('vumeterprocessor.js');
 </pre>
-            <h4>
-              AudioWorkletGlobalScope: vumeterprocessor.js
-            </h4>
-            <pre>
-var VU_METER_SMOOTHING = 0.9;
-var VU_METER_MINIMUM = 0.00001;
+            <pre class="example" title=
+            "VUMeterNode - AudioWorkletGlobalScope (vumeterprocessor.js)">
+registerAudioWorkletProcessor('VUMeter', class extends AudioWorkletProcessor {
 
-registerProcessor('VUMeter', class extends AudioWorkletProcessor {
-
-  // Note: AudioParam definition can be omitted.
+  static meterSmoothingFactor = 0.9;
+  static meterMinimum = 0.00001;
 
   constructor (options) {
     super(options);
+    this._volume = 0;
+    this._updatingInterval = options.updatingInterval;
+    this._nextUpdateFrames = this.interval;
 
-    this.volume = 0;
-    this.updatingInterval = options.updatingInterval;
-    this.nextUpdateFrames = this.interval;
+    this.onmessage = event =&gt; {
+      if (event.data.hasOwnProperty('updatingInterval'))
+        this._updatingInterval = event.data.updatingInterval;
+    }
   }
 
   get interval () {
-    return this.updatingInterval / 1000 * this.contextInfo.sampleRate;
+    return this._updatingInterval / 1000 * this.contextInfo.sampleRate;
   }
 
   process (inputs, outputs, parameters) {
-    // Note that the input will be downmixed to mono; however,
-    // if no inputs are connected then zero channels will be passed in.
-    if (inputs[0].length &gt; 0) {
-      var buffer = inputs[0][0];
-      var bufferLength = buffer.length;
-      var sum = 0, x = 0, rms = 0;
-
-      // Calculated the squared-sum.
-      for (var i = 0; i &lt; bufferLength; ++i) {
-        x = buffer[i];
-        sum += x * x;
-      }
+    // Note that the input will be downmixed to mono; however, if no inputs are
+    // connected then zero channels will be passed in.
+    let buffer = inputs[0][0];
```

Comment:
I think `inputs[0].length > 0` is checking the channel count of the first input. Is this what we want? Shouldn't it be checking `inputs.length > 0'?